kubemarine/procedures/check_iaas.py

```diff
@@ -1372,6 +1372,55 @@ def check_ipip_tunnel(group: NodeGroup) -> Set[str]:
                               f"sudo rm -f {ipip_check} {ipip_check}.pid", warn=True)
 
 
+def fs_mount_options(cluster: KubernetesCluster) -> None:
+    with TestCase(cluster, '018', 'System', 'Filesystem mount options') as tc,\
+            suspend_firewalld(cluster):
+
+        failed_nodes: Set[str] = set()
+        # Only Kubernetes nodes should be checked
+        group = cluster.make_group_from_roles(['control-plane', 'worker']).get_sudo_nodes()
+        collector = CollectorCallback(group.cluster)
+        # Get CRI root
+        if cluster.inventory['services']['cri']['containerRuntime'] == "containerd":
+            # Containerd root
+            if cluster.inventory['services']['cri']['containerdConfig'].get('root', ""):
+                cri_root = cluster.inventory['services']['cri']['containerdConfig']['root']
+            else:
+                cri_root = "/var/lib/containerd"
+        # Get rid of that part after KubeMarine stop supporting Kubernetes version lower v1.24
+        if cluster.inventory['services']['cri']['containerRuntime'] == "docker":
+            # Docker root
+            if cluster.inventory['services']['cri']['dockerConfig'].get('data-root', ""):
+                cri_root = cluster.inventory['services']['cri']['dockerConfig']['data-root']
+            else:
+                cri_root = "/var/lib/docker"
+        cluster.log.debug("Mount options check")
+        # Check the mount options for filesystem where containerd root is located.
+        # If containerd root doesn't exist the script check the parent directory and so forth.
+        # At the end of the script 'findmnt' return the filesytem mount point, device,
+        # and mount options for nearest parent directory of containerd root.
+        # 'findmnt' perform the recursive search of mount point for filesystem
+        # from the given path to the root, that's exactly what we need.
+        cmd = f"CRI_PATH={cri_root}; while [ ! -d \"${{CRI_PATH}}\" ]; do CRI_PATH=$(dirname \"${{CRI_PATH}}\"); " \
+              f"done; findmnt -T \"${{CRI_PATH}}\" | grep 'nosuid'"
+        with group.new_executor() as exe:
+            for node_exe in exe.group.get_ordered_members_list():
+                node_exe.run(f"{cmd}", warn=True, callback=collector)
+
+        # Check output and create result message
+        for host, item in collector.result.items():
```

Comment:
In case the command does not depend on node, it is simpler to just NodeGroup.run
```suggestion
        result = group.run(f"{cmd}", warn=True)

        # Check output and create result message
        for host, item in result.items():
```

Reply:
Yes, we could simplify it.